### PR TITLE
Add a multi-stage Docker build for production apps

### PR DESCRIPTION
**Pull request by eatyourgreens**
Package:
lib-classifier
app-project
app-content-pages

- Split the docker build into `bootstrap`, which installs the monorepo and builds dependencies, and `production-apps`, which builds the NextJS apps.
- Remove the NextJS builds from the bootstrap script.
- update the docker-compose setup to use the appropriate named target for each local docker image. `front-end-monorepo_dev` builds from the `bootstrap` stage. `front-end-monorepo_prod` builds from the `production-apps` stage.
- rename the test service to yarn, since it runs yarn commands, and remove local mounted volumes when testing the production set-up. If local files change, you'll need to run `docker-compose build` to test the changes in a new production image.
- Update the Jenkins build to build `zooniverse/front-end-monorepo` with the bootstrapped monorepo, then build and start the NextJS apps in their respective images.

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && yarn bootstrap` and app works as expected?
- [ ] Can you run a [production build](https://github.com/zooniverse/front-end-monorepo#getting-started) of the app?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
 RUN yarn install
 
 RUN yarn bootstrap
+
+FROM node:10 AS production-apps
+
+RUN mkdir -p /usr/src
+WORKDIR /usr/src/
+
+COPY --from=bootstrap /usr/src /usr/src
```
Comment (eatyourgreens) [inline Dockerfile]: Ideally, this step should only copy over the files that are needed to run the production apps, not the entire source tree, tests etc.

**Comment by eatyourgreens** [comment]
Speed, really, but I think I need feedback from people who have used to this to find out if it’s useful.

**Comment by eatyourgreens** [comment]
Bumping this because it's almost a month old and should fix the duplicate production builds on Travis CI.

**Comment by rogerhutchings** [comment]
Just for reference, a first build with `--no-cache` took a brutal 23 minutes 😲 but it'll do that on master too. subsequently starting the project app was a bit snappier

**Pull request merged**